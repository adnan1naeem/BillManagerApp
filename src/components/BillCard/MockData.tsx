const MockData = [
  {
    month: "Jan",
    budget: 2000,
    name:"Bills"
  },
  {
    month: "Jan",
    budget: 2000,
    name:"Bills"
  },{
    month: "Jan",
    budget: 2000,
    name:"Bills"
  }
  
]



