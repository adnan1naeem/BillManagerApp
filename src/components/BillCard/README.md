#  BillCard Component

## Compatibility

| 🌏 Web                   | 🖥 Electron              | 📱 React Native |
| :----------------------: | :----------------------: | :-------------: |
| :heavy_multiplication_x: | :heavy_multiplication_x: | ::              |

## Screenshots

| 🌏 Web | 🖥 Electron | 📱 React Native                         |
| :----: | :---------: | :-------------------------------------: |
| TBD    | TBD         | ![BillCard](./Screenshots/BillCard.png) |

## Props

| Name   | Type   | Default | Description                           |
| :----- | :----- | :------ | :------------------------------------ |
| name   | string |         | It is used for adding billName        |
| month  | string |         | It is used for adding month of bill   |
| budget | number |         | It is used for adding budget of bills |
| style  | object |         | It is used for adding stylesin AppBar |


## How to use

```react
import React from 'react';
<BillCard
      name={"ADNAN "}
      month={"Jan"}
      budget={2000}
    />
```

## Tests

