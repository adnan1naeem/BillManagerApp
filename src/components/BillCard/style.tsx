export const style: any = {
	card: {

	},
	link: {
		textDecoration: 'none'
	},
	text: {
		fontSize: '14px',
		overflow: 'hidden',
		textOverflow: 'ellipsis',
		whiteSpace: 'nowrap',
		display: 'block',
		color: '#373a3c'
	},
	container: {
		width: 150,
		paddingTop: 10,
		paddingBottom: 10
	},
	body: {
		display: 'flex',
		padding: 7,
		flexDirection: 'column',
		backgroundColor: 'white'
	},

	img: {
		height: 180,

		opacity: '0.2'
	},
	cardBody: {
		display: 'flex',
		flexDirection: 'column',
	},
	topView: {
		height: 180,
		backgroundColor: '#0093FF',
		display: 'flex',
		flexDirection: 'column',
		padding: 7
	},
	consumedLabel: {
		color: 'white',
		fontSize: 12,
	},
	budgetContanier: {
		display: 'flex',
		flexDirection: 'row'
	},
	budget: {
		color: 'white',
		fontSize: 20,
		flex: 1
	},


	name: {
		color: '#0093FF',
		fontSize: 14,
		overflow: 'hidden',
	},

	month: {
		color: '#0093FF',
		fontSize: 12,
		flex: 1

	},
};