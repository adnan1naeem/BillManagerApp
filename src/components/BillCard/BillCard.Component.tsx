import React from 'react'
import View from "../View/View.Component";
import Text from "../Text/Text.Component";
import { style } from "./style"

export interface BillCardProperties {
	name: string,
	month: string,
	budget: number,
	Style?: any,
}

const BillCard =(props: BillCardProperties) => {

	const { name, month, budget } = props;

	return (
		<View style={style.container}>
			<View style={style.topView}>
				<View style={style.descriptionView}>
					<Text style={style.consumedLabel}>
						Total Consumed
					</Text>
					<View style={style.budgetContanier}>
						<Text style={style.budget}>
							Rs. {budget}
						</Text>
					</View>
				</View>
			</View>
			<View style={style.body}>
				<View style={style.lowerView}>
					<Text style={style.name}>
						{name}
					</Text>
					<View style={style.monthContainer}>
						<Text style={style.month}>
							{month}
						</Text>
					</View>
				</View>
			</View>
		</View>
	);
};

export default BillCard;