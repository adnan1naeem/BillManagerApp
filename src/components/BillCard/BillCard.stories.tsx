import React from 'react';
import { storiesOf } from '@storybook/react-native';
import BillCard from './BillCard.component';
import { withKnobs} from '@storybook/addon-knobs/react';
import {View} from 'react-native';


storiesOf('Card', module)
  .addDecorator(getStory => <View>{getStory()}</View>)
  .addDecorator(withKnobs )
  .add('Card Story', () => (
  <BillCard  
  name="ADNAN"
	month="June"
	budget={22222}
  />
  ))

  