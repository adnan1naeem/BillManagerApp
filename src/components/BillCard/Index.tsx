import BillCard from  "./BillCard.component";
export default BillCard; 