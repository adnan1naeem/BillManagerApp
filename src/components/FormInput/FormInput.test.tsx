import * as React from 'react'
import { shallow,configure } from 'enzyme';
import BillCard from  "./BillCard.component";
import ReactSixteenAdapter from 'enzyme-adapter-react-16';

// it('renders without crashing', () => {
//   const rendered = renderer.create(<App />).toJSON()
//   expect(rendered).toMatchSnapshot()
//   expect(rendered).toBeTruthy()
// })
configure({ adapter: new ReactSixteenAdapter() })


describe('BillCard Testcases', () => {
  it('should render component', () => {
   expect(<BillCard name="Bills" month="Jan" budget={2000} />).toBeTruthy();
  });

  it('should throw error as required prop Budget is missing', () => {
    expect(()=>    <BillCard name="Bills" month="Jan"  /> ).toThrow()
   });
});