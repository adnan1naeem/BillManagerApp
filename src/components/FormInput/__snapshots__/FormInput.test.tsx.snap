// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormPicker Input unit Tests FormPickerInput SnapShot Test 1`] = `
<View
  style={
    Object {
      "display": "flex",
      "flex": 1,
      "flexDirection": "column",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "marginLeft": 20,
      }
    }
  >
    label
  </Text>
  <View
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    />
    <TextInput
      allowFontScaling={true}
      label="label"
      onChangeText={undefined}
      style={
        Object {
          "borderBottomColor": "#4381FF",
          "color": "#4381FF",
          "flex": 1,
          "height": 30,
          "width": 300,
        }
      }
      value="Name"
    />
  </View>
</View>
`;
